src/main/java/edu/umich/its/spe/SPEEsbImpl.java

```diff
@@ -105,7 +105,8 @@ public WAPIResultWrapper getGradesVia(SPEProperties speproperties, String graded
 			.append("/Unizin/data/CourseId/")
 			.append(values.get("COURSEID"))
 			.append("/AssignmentTitle/")
-			.append(URLEncoder.encode(values.get("ASSIGNMENTTITLE"),"UTF-8"));
+			.append(URLEncoder.encode(values.get("ASSIGNMENTTITLE"),"UTF-8").replaceAll("\\+", "%20")
```

Comment:
Add a comment line here as "The url to query the ESB must encode blank spaces as %20 not as +. "?